From aaa8662ff56fe2818e7d63b91792d0a706b3ad38 Mon Sep 17 00:00:00 2001
From: jkreierh <jkreierh@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 28 Jan 2016 13:24:02 +0000
Subject: [PATCH 0729/1077] MANTIS 39795 Mantis 39795: CCM Conveyor Device
 Handler Extend "ConveyorDev.idle" to supply baggage placement data
 (dimensions and error codes)

git-svn-id: svn://localhost/SelfServiceCommon/trunk@972 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/ConveyorDev.idl         | 47 ++++++++++++++++---
 1 file changed, 40 insertions(+), 7 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
index f059ca5d..c3a507a8 100644
--- a/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/ConveyorDev.idl
@@ -39,7 +39,7 @@
 **/
 module conveyordev
 {
-   const string interface_version = "1.7";
+   const string interface_version = "1.8";
 /*
 using DeviceComponents::indicationListener and DeviceComponents::Generic from mdci.idl
   RC indicationListener::indication(in any ind);
@@ -166,12 +166,19 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       ERR_TRANSPORT_FAILED,         /**< piece transport failed. */
       ERR_PIECE_TOO_HEAVY,          /**< Piece is too heavy. */
       ERR_TRANSPORT_ABORTED,        /**< Piece transport not started. */
-      ERR_EMERGENCY_STOP,           /**< conveyor emergency stop button pressed */
-      ERR_USER_INTERFERENCE,        /**< user interference which are _not_ considered
+      ERR_EMERGENCY_STOP,           /**< Conveyor emergency stop button pressed */
+      ERR_USER_INTERFERENCE,        /**< User interference which are _not_ considered
                                          critical to health, safety and security */
-      ERR_SAFETY_INTRUSION,         /**< safety intrusions which are considered
+      ERR_SAFETY_INTRUSION,         /**< Safety intrusions which are considered
                                          critical to health, safety and security */
-      ERR_THERMAL_OVERLOAD          /**< too hot to work */
+      ERR_THERMAL_OVERLOAD,         /**< Too hot to work. */
+      ERR_PIECE_RESTLESS,           /**< Could be an animal inside. */
+      ERR_INCORRECTLY_PLACED,       /**< Piece placed in the wrong way. */
+      ERR_TUB_REQUIRED,             /**< The shape of the bag requires a tub. */
+      ERR_HANDLE_EXTENDED,          /**< A handle protrudes from the bag. */
+      ERR_PIECE_TOO_WIDE,           /**< Piece is to wide. */
+      ERR_PIECE_OVERSIZED,          /**< Piece is too big. */
+      ERR_WEIGHT_OUT_OF_RANGE       /**< Piece is too light or too heavy. */
    };
 
    /**
@@ -312,7 +319,32 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
        * Additionally delivery notes are sent here.
        *
       **/
-      StringList  baggageBarcodes;
+      StringList        baggageBarcodes;
+
+      /**
+       *
+       * The height of the current piece in millimeter.
+       * Use -1 to indicate, that no measurement has been taken.
+       *
+      **/
+      long              baggageHeight;
+
+      /**
+       *
+       * The length of the current piece in millimeter.
+       * Use -1 to indicate, that no measurement has been taken.
+       *
+      **/
+      long              baggageLength;
+
+      /**
+       *
+       * The width of the current piece in millimeter.
+       * Use -1 to indicate, that no measurement has been taken.
+       *
+      **/
+      long              baggageWidth;
+
    };
 
    typedef sequence<octet> image;
@@ -400,7 +432,8 @@ using DeviceComponents::indicationListener and DeviceComponents::Generic from md
       IND_PARKING_RETURN_DONE,               /**< Finished CMD_PARKING_RETURN */
       IND_PIECE_IMAGE,                       /**< image(s) of the piece. */
       IND_PIECE_NOREAD,                      /**< No data read form any label. */
-      IND_PIECE_DELIVERED                    /**< Piece has left kiosk-controlled conveyors. */
+      IND_PIECE_DELIVERED,                   /**< Piece has left kiosk-controlled conveyors. */
+      IND_PIECE_DIMENSIONS                   /**< Piece-height, length, or width. */
    };
 
    /**
-- 
2.41.0.windows.1

